tests/complexity/test_memory.py

```diff
@@ -0,0 +1,214 @@
+            2 * READ + WRITE,# tanh_real_scalar
+            2 * READ + WRITE,# arcsinh_real_scalar
+            2 * READ + WRITE,# arccosh_real_scalar
+            2 * READ + WRITE,# arctanh_real_scalar
+            3 * READ + WRITE,# arctan2_real_scalar
+            (READ + WRITE) * SHAPE(out, 0),## sin_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## cos_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## tan_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arcsin_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arccos_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arctan_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## sinh_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## cosh_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## tanh_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arcsinh_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arccosh_real_array_1d
+            (READ + WRITE) * SHAPE(out, 0),## arctanh_real_array_1d
+            (2 * READ + WRITE) * SHAPE(out, 0),## arctan2_real_array_1d
+            4 * READ + WRITE,# numpy_math_expr_real_scalar
+            (3 * READ + WRITE) * SHAPE(out, 0)## numpy_math_expr_real_array_1d
+            ]
+
+
+    for f, c in complexity.costs.items():
+        assert sp_simplify(c - comp[i]) == 0
+        i = i + 1
+
+
+
+
+
+# ==============================================================================
+def test_complexity_ex2():
+
+    f = path_dir + '/ex2.py'
+    mode = None
+
+    complexity = MemComplexity(f)
+    complexity.cost(mode=mode, simplify=True, bigo=None)
+#    complexity.cost(mode=mode, simplify=True, bigo=['n'])
+
+    print('----------------------')
+    i = 0
+    comp = [READ + WRITE + n**2*(3*READ*(n - 1) + 8*READ + WRITE*(n - 1) + 6*WRITE)/2,# f1
+            READ + WRITE + n ** 2 * (4 * READ * (n - 1) + 6 * READ + WRITE * (n - 1) + 4 * WRITE) / 2,# f2
+            READ * n ** 3 / 2 + 3 * READ * n ** 2 / 2 + READ + WRITE * n ** 3 / 6 + 3 * WRITE * n ** 2 / 2 + WRITE * n / 3 + WRITE# f3
+            ]
+
+
+    for f, c in complexity.costs.items():
+        assert sp_simplify(c - comp[i]) == 0
+        i = i + 1
+
+
+
+# ==============================================================================
+def test_complexity_ex_assembly():
+
+    f = path_dir + '/ex_assembly.py'
+    mode = None
+
+    complexity = MemComplexity(f)
+    complexity.cost(mode=mode, simplify=True, bigo=None)
+#    complexity.cost(mode=mode, simplify=True, bigo=['n'])
+
+    print('----------------------')
+    i = 0
+    comp = [2*READ + WRITE*k1*k2 + WRITE*(p1 + 1)*(p2 + 1) + 2*WRITE + ne1*(READ + WRITE + ne2*(READ + WRITE*k1*k2 + WRITE + (p1 + 1)**2*(p2 + 1)**2*(14*READ + 6*WRITE + k1*k2*(18*READ + 7*WRITE)) + (p1 + 1)*(p2 + 1)*(READ*(p1 + 1)*(p2 + 1) + WRITE) + (p1 + 1)*(p2 + 1)*(READ + WRITE + k1*(READ + WRITE + k2*(5*READ + 2*WRITE)))))]
```

Comment:
See above, function name please. Also could you please break this line up so it fits on a screen please